diff --git a/spyder/plugins/application/widgets/status.py b/spyder/plugins/application/widgets/status.py
index 80c89caf0..adf89fada 100644
--- a/spyder/plugins/application/widgets/status.py
+++ b/spyder/plugins/application/widgets/status.py
@@ -9,8 +9,10 @@
 # Standard library imports
 import os.path as osp
 import datetime
+import webbrowser
 
 # Third-party imports
+from qtpy import QtModuleNotInstalledError
 from qtpy.QtCore import Qt, QUrl
 from qtpy.QtWidgets import QDialog, QVBoxLayout
 
@@ -23,7 +25,6 @@
 from spyder.utils.icon_manager import ima
 from spyder.utils.qthelpers import start_file
 from spyder.utils.stylesheet import WIN
-from spyder.widgets.browser import WebView
 
 
 class InAppAppealDialog(QDialog, SpyderFontsMixin):
@@ -52,6 +53,7 @@ def __init__(self, parent=None):
             "index.html",
         )
 
+        from spyder.widgets.browser import WebView
         # Create webview to render the appeal message
         webview = WebView(self, handle_links=True)
 
@@ -116,11 +118,17 @@ def _on_click(self):
     # ---- Public API
     # -------------------------------------------------------------------------
     def show_appeal(self):
-        if self._appeal_dialog is None:
-            self._appeal_dialog = InAppAppealDialog(self)
-
-        if not self._appeal_dialog.isVisible():
-            self._appeal_dialog.show()
+        try:
+            if self._appeal_dialog is None:
+                self._appeal_dialog = InAppAppealDialog(self)
+
+            if not self._appeal_dialog.isVisible():
+                self._appeal_dialog.show()
+        except QtModuleNotInstalledError:
+            # QtWebEngineWidgets is optional, so just open the URL in the
+            # default browser.
+            # See spyder-ide/spyder#24905 for the details.
+            webbrowser.open("https://www.spyder-ide.org/donate")
 
     # ---- StatusBarWidget API
     # -------------------------------------------------------------------------
