diff --git a/spyder/app/start.py b/spyder/app/start.py
index 3f225a164..2ad20aa5f 100644
--- a/spyder/app/start.py
+++ b/spyder/app/start.py
@@ -44,7 +44,8 @@
 
 # Prevent a race condition with ZMQ
 # See spyder-ide/spyder#5324.
-import zmq  # noqa
+if not sys.platform == 'darwin':
+    import zmq  # noqa
 
 # Load GL library to prevent segmentation faults on some Linux systems
 # See spyder-ide/spyder#3226 and spyder-ide/spyder#3332.
